graph.go

```diff
@@ -0,0 +1,293 @@
+package dalec
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/pmengelbert/stack"
+	"golang.org/x/exp/constraints"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type Graph struct {
+	target   string
+	specs    map[string]*Spec
+	ordered  orderedDeps
+	indices  map[string]int
+	vertices []*vertex
+	edges    sets.Set[dependency]
+}
+
+type dependency struct {
+	v    *vertex
+	w    *vertex
```

Comment:
slight nit but can we rename these to either `v1' and `v2' or `u` and 'v'? 'u' and 'v' are the most standard for notating an edge, in my experience

Reply:
done